testplan/testing/multitest/logging.py

```diff
@@ -0,0 +1,166 @@
+from __future__ import absolute_import
+
+import logging
+from collections import namedtuple
+from contextlib import contextmanager
+from six import StringIO
+from logging import StreamHandler
+from tempfile import NamedTemporaryFile
+
+from testplan.common.utils.logger import _LOGFILE_FORMAT, Loggable
+
+CAPTURED_LOG_DESCRIPTION = "Auto Captured Log"
+
+
+class CaptureLevel(object):
+    """Capture level Enum like object
+
+    ROOT:
+        Capture all logs reaching the root logger, it contains all testplan logs plus other lib logs
+    TESTPLAN:
+        Capture all testplan logs, eg driver logs
+    TESTSUITE:
+        Whatever is logged from the testcases
+    """
+
+    ROOT = staticmethod(lambda suite: logging.getLogger())
+    TESTPLAN = staticmethod(lambda suite: suite.logger.parent)
+    TESTSUITE = staticmethod(lambda suite: suite.logger)
+
+
+class LogCaptureConfig(object):
+    """
+    Configuration for log capture
+
+    Attributes
+    ----------
+    capture_level CaptureLevel:
+        initial value: CaptureLevel.TESTSUITE The level the log are captured, TESTSUITE (default), TESTPLAN or ROOT
+    attach_log bool:
+        If True the logs captured to file and then attached to the result
+    format str:
+        A format string can be passed to the loghandler
+
+
+    """
+
+    def __init__(self):
+        self.capture_level = CaptureLevel.TESTSUITE
+        self.attach_log = False
+        self.format = _LOGFILE_FORMAT
+
+
+class LogCaptureMixin(Loggable):
+    """ Mixin to add easy logging support to any @multitest.testsuite
+
+    """
+
```

Comment:
hmm, i thought black would complain about this docstring...

Reply:
I will fix this to a proper one liner